uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Document.php

```diff
@@ -12,6 +12,14 @@ public static function create(&$params)
     {
         $entityName = 'Document';
         $hook = empty($params['id']) ? 'create' : 'edit';
+
+        if ($hook === 'create') {
+            // If creating a new Document, we require all three contacts to be defined.
+            if (empty($params['source_contact_id'] || empty($params['target_contact_id']) || empty($params['assignee_contact_id']))) {
+                throw new CiviCRM_API3_Exception("Please specify 'source_contact_id', 'target_contact_id' and 'assignee_contact_id'.");
+            }
```

Comment:
If we forget what I said above for a while ... you can refactor the code to this :

```php
if ($hook === 'create') {
  // If creating a new Document, we require all three contacts to be defined.
  if (empty($params['source_contact_id'] || empty($params['target_contact_id']) || empty($params['assignee_contact_id']))) {
    throw new CiviCRM_API3_Exception("Please specify 'source_contact_id', 'target_contact_id' and 'assignee_contact_id'.");
  }
  if (empty($params['status_id']))
  {
    $params['status_id'] = 1;
  }
}

 CRM_Utils_Hook::pre($hook, $entityName, CRM_Utils_Array::value('id', $params), $params);
```

same apply for /Tasksassignments/BAO/Task.php